its/ruling/src/test/resources/guava/java-S3457.json

```diff
@@ -0,0 +1,25 @@
+{
+'com.google.guava:guava:src/com/google/common/io/Closer.java':[
```

Comment:
All these 9 issues are FPs. The issue message says: `Format specifiers should be used instead of string concatenation.`, but the method which is called is: `java.util.logging.Logger.log(Level, String, Throwable)`.

Consequently, you HAVE to use string concatenation

Reply:
:tired_face: My understanding of `java.util.logging.Logger` was wrong.
I thought that:
```java
LOG.log(Level.WARNING, "Message: " + param, ex);
```
Is equivalent to:
```java
LOG.log(Level.WARNING, "Message: {0}", new Object[] {param, ex});
```
But, I tested it and it's not like slf4j. The last parameter of `log(Level level, String msg, Object params[])` is not handled like `thrown` in `log(Level level, String msg, Throwable thrown)`.

In conclusion, I also think these are FPs.

Reply:
So it was 9 FPs, because of the wrong message: `Format specifiers should be used instead of string concatenation`
For those 9 issues, I changed the message to: `Lambda should be used to differ string concatenation.`
For example, in case of `src/com/google/common/io/Closer.java:252`:
```
      Closeables.logger.log(Level.WARNING,
          "Suppressing exception thrown when closing " + closeable, suppressed);
```
Can be changed into:
```
      Closeables.logger.log(Level.WARNING,
          suppressed, () -> "Suppressing exception thrown when closing " + closeable);
```

Reply:
I have a question about your new message, "Lambda should be used to differ string concatenation."  It makes no sense as written.

"Differ" means "is different from," as in "This thing differs from that thing because it is orange."

"Defer" means delay or postpone, as in "Let us defer that decision until next week."

Do you mean "Lambda should be used to **postpone/delay** string concatenation"?

Reply:
Indeed @FredWRobinson, that's the second option which is correct. It's a typo. I created the following ticket to track the change: [SONARJAVA-3587](https://jira.sonarsource.com/browse/SONARJAVA-3587)

Note that we are not really following messages on closed pull requests. If you want to be sure to reach us, don't hesitate to use https://community.sonarsource.com/

Cheers,
Michael